[#1202] Missing javadoc for functions

yhtMinceraft1010X opened the pull request:
Fixes #1202.

Proposed Commit Message:
```
Some non-trivial private methods have no Javadoc. At the same time, 
Javadoc styles vary across the files. This is also due to some Javadoc
checks not being used in Checkstyle.

Let's add the missing Javadoc, standardize the style and implement
additional checks in Checkstyle.
```

Proposed Javadoc standard for RepoSense (in addition to what is already covered in [SE-EDU](https://se-education.org/guides/conventions/java/index.html#comments) and [Google](https://google.github.io/styleguide/javaguide.html#s7-javadoc)) can be found in:
* Raw `styleGuides.md` file: https://github.com/reposense/RepoSense/pull/1650/files#diff-bd63b2861c8a46fbd8a67f313e555901c8d6d06762d1c0551c3254467f45b484
* Style Guides on `surge.sh`: https://docs-1650-pr-reposense-reposense.surge.sh/dg/styleGuides.html

dcshzj commented:
I'll let @reposense/active-developer review first.

yhtMinceraft1010X commented:
Requesting review from @reposense/active-developer as per [week 5 schedule](https://nus-cs3281.github.io/website/schedule/index-cs3281.html)

chan-j-d commented on src/main/java/reposense/git/GitClone.java:
```diff
@@ -94,6 +95,8 @@ public static void cloneShallowPartialBare(RepoConfiguration config, Path rootPa
     /**
      * Clones repo specified in the {@code config} and updates it with the branch info.
+     *
+     * @throws GitCloneException when an error occurs during command execution.
```
I think this description can be more explicit. Maybe something like `GitCloneException when an error occurs while attempting to clone the repo`. Though I realise this exact line is used multiple times over the entire file so if you were to change this, might have to change the rest as well.

The wording currently used sounds it would fit a more generic `CommandRunnerProcessException` instead.

yhtMinceraft1010X replied:
Thanks for pointing it out. I have made the change accordingly.

chan-j-d commented on src/main/java/reposense/git/GitShortlog.java:
```diff
@@ -36,6 +36,10 @@
     }
 
+    /**
+     * Obtains summarised version of git log command for the date range given by {@code sinceDate}
```
Can consider specifying that it runs this command against the repository located at `root`. Something like `Obtains summarised version of git log from the repository at {@code root} for the date range...`

yhtMinceraft1010X replied:
Fixed

chan-j-d commented on src/main/java/reposense/git/GitShow.java:
```diff
@@ -48,7 +50,11 @@ public static CommitHash getExpandedCommitHash(String root, String shortCommitHa
+     *
+     * @throws CommitNotFoundException if no commit exists for the given {@code commitHash}.
+     * @throws ParseException if the date string for the given {@code commitHash} could not be parsed into
+     * a {@code Date}.
```
I think it can be specified as 'could not be parsed into a {@code Date} _object_.' in order to highlight the difference between that and the date string.

yhtMinceraft1010X replied:
Fixed

chan-j-d commented on src/main/java/reposense/model/AuthorConfiguration.java:
```diff
@@ -69,9 +69,11 @@ public void update(StandaloneConfig standaloneConfig, List<String> ignoreGlobLis
-     * Checks for duplicate aliases in {@code authorDetailsToAuthorMap} and generates warnings
+     * Checks for duplicate aliases in {@code authorDetailsToAuthorMap} and generates warnings.
+     *
      * @param authorDetailsToAuthorMap
```
Can consider adding 'Map to be checked for duplicate aliases.`

yhtMinceraft1010X replied:
Fixed

chan-j-d commented on src/main/java/reposense/report/ReportGenerator.java:
```diff
@@ -172,7 +173,9 @@ private static void setReportConfiguration(ReportConfiguration config, String ou
-     * Set title of template file located at {@code filePath} to {@code pageTitle}
+     * Set title of template file located at {@code filePath} to {@code pageTitle}.
+     *
+     * @throws IOException if file writing fails.
```
I think 'if writing to the file fails' would be better.

yhtMinceraft1010X replied:
Fixed.

chan-j-d commented on src/main/java/reposense/util/FileUtil.java:
```diff
@@ -229,6 +237,8 @@ public static void copyDirectoryContents(String sourcePath, String outputPath, L
     /**
      * Creates the {@code dest} directory if it does not exist.
+     *
+     * @throws IOException for any I/O errors in the underlying function.
```
I think a better description would be something like `IOException if the directory could not be created.` I believe there is no need to describe where the errors come from.

yhtMinceraft1010X replied:
Fixed.

chan-j-d commented on src/systemtest/java/reposense/ConfigSystemTest.java:
```diff
@@ -133,8 +133,8 @@ private String getInputWithDates(String sinceDate, String untilDate) {
     /**
-     * Generates the testing report and then compared it with the expected report
-     * Re-generates a normal report after the testing finished if the first report is shallow-cloned
+     * Generates the testing report and then compares it with the expected report.
```
I think the 'then' in 'then compares' can be dropped.

yhtMinceraft1010X replied:
Fixed.

chan-j-d reviewed (changes requested):
Thanks for updating the Javadocs. Must have been a lot of work. Most of my points are quite minor besides a few.

yhtMinceraft1010X commented on src/main/java/reposense/model/AuthorConfiguration.java:
```diff
@@ -69,9 +69,11 @@ public void update(StandaloneConfig standaloneConfig, List<String> ignoreGlobLis
 
     /**
-     * Checks for duplicate aliases in {@code authorDetailsToAuthorMap} and generates warnings
+     * Checks for duplicate aliases in {@code authorDetailsToAuthorMap} and generates warnings.
```
Fixed

yhtMinceraft1010X commented on src/main/java/reposense/util/FileUtil.java:
```diff
@@ -306,6 +316,8 @@ public static boolean isEmptyFile(String directoryPath, String relativeFilePath)
     /**
      * Returns a list of {@code Path} of {@code fileTypes} contained in the given {@code directoryPath} directory.
+     *
+     * @throws IOException if an I/O error occurs while accessing {@code directoryPath}.
```
Fixed

yhtMinceraft1010X commented on src/main/java/reposense/util/TimeUtil.java:
```diff
@@ -210,6 +210,8 @@ public static String extractDate(String date) {
      * Setting setLenient to false prevents unexpected results.
      * Without it, even with "dd/MM/yyyy HH:mm:ss" format, 11/31/2017 00:00:00 will be parsed to 11/7/2019 00:00:00.
+     *
+     * @throws java.text.ParseException if date cannot be parsed for the given format.
```
Fixed.

chan-j-d commented:
Looks good to me! Requesting review from @dcshzj.

dcshzj commented on src/main/java/reposense/parser/CsvParser.java:
```diff
@@ -56,7 +56,10 @@
      * @throws IOException if {@code csvFilePath} is an invalid path.
+     * @throws FileNotFoundException if the csv file cannot be found in the provided {@code csvFilePath}.
      */
     public CsvParser(Path csvFilePath) throws IOException {
```
I think we should add a `throws FileNotFoundException` into the method signature.

dcshzj commented on src/main/java/reposense/util/FileUtil.java:
```diff
@@ -91,6 +91,7 @@ public static void zipFoldersAndFiles(List<Path> pathsToZip,
      * Writes the JSON file representing the {@code object} at the given {@code path}.
+     *
      * @return An Optional containing the Path to the JSON file, or an empty Optional
      *         if there was an error while writing the JSON file.
```
I think this line should not have indentation.

dcshzj reviewed (changes requested):
Some additional files that needs to be modified:
1. GitDiff.java at line 28 (violates the additional conventions you proposed)
2. GitLog.java at line 40 and line 57 (violates the additional conventions you proposed)
3. RepoConfiguration.java at line 126, line 232 and line 247 (violates the additional conventions you proposed)
4. RepoConfiguration.java at line 366-367 and line 374-375 (can be merged into a single line)
5. AuthorConfigCsvParser.java at line 59 (violates the additional conventions you proposed)
6. CsvParser.java at line 205 (extra newline)
7. CsvParser.java at line 207 (extra space between `return` and `record`)
8. RepoConfigCsvParser.java at line 66 (missing an `@throws InvalidLocationException`)
9. SinceDateArgumentType.java at line 26 (missing an `@throws ArgumentParserException`) [maybe not needed, I'm not sure, I leave this up to you to see if it is necessary]
10. RepoCloner.java at line 118 (violates the additional conventions you proposed)
11. RepoCloner.java at line 202 (incorrect indentation)
12. ReportGenerator.java at line 284 (incorrect indentation)

yhtMinceraft1010X commented:
I would like to thank @chan-j-d for explaining the methods in `AnnotatorAnalyzer.java`, which helped when refining the Javadoc for them.

yhtMinceraft1010X commented on src/main/java/reposense/git/GitUtil.java:
```diff
@@ -75,7 +76,7 @@ static String convertToFilterAuthorArgs(Author author) {
+     * Returns the {@code String} command to specify the file {@code formats} to analyze for `git` commands.
      */
     public static String convertToGitFormatsArgs(List<FileType> formats) {
         StringBuilder gitFormatsArgsBuilder = new StringBuilder();
```
Thanks for pointing this out. I have fixed it

yhtMinceraft1010X commented on src/main/java/reposense/model/FileType.java:
```diff
@@ -51,6 +51,7 @@ public static FileType convertStringFormatToFileType(String format) {
 
```
Fixed

dcshzj commented on src/main/java/reposense/authorship/FileInfoAnalyzer.java:
```diff
@@ -135,7 +137,7 @@ private static FileResult generateBinaryFileResult(RepoConfiguration config, Fil
     /**
      * Sets the {@code Author} and {@code Date} for each line in {@code fileInfo} based on the git blame analysis
-     * on the file.
+     * on the file and {@code config}.
```
This comment does not really make sense. We are not doing a git blame analysis on config, consider rephrasing this sentence?

dcshzj commented on src/main/java/reposense/authorship/FileInfoExtractor.java:
```diff
@@ -238,7 +241,7 @@ private static int getStartingLineNumber(String linesChanged) {
 
     /**
-     * Returns true if {@code filePath} is valid and the file is not in binary.
+     * Returns true if {@code filePath} is valid and the file is not in binary (I.e. part of {@code textFilesSet}).
```
```suggestion
     * Returns true if {@code filePath} is valid and the file is not in binary (i.e. part of {@code textFilesSet}).
```

dcshzj commented on src/main/java/reposense/authorship/FileInfoExtractor.java:
```diff
@@ -187,6 +187,9 @@ private static void setLinesToTrack(FileInfo fileInfo, String fileDiffResult) {
      * Traverses each file from the repo root directory, generates the {@code FileInfo} for each relevant file found
      * based on {@code config} and inserts it into {@code fileInfos}.
+     * <p>
+     * Adds binary files to {@code List<FileInfo>} if {@code isBinaryFiles} is true. Else, adds non-binary files
```
`Else` should change to `Otherwise`.

dcshzj commented on src/systemtest/java/reposense/ConfigSystemTest.java:
```diff
@@ -133,8 +133,16 @@ private String getInputWithDates(String sinceDate, String untilDate) {
+     * @throws Exception if any occur during testing.
      */
     private void runTest(String inputDates, boolean shouldIncludeModifiedDateInLines,
                         boolean shallowCloning, boolean shouldFreshClone, boolean findPreviousAuthors,
```
This line and the next line should have the indentation fixed

dcshzj commented on src/systemtest/java/reposense/ConfigSystemTest.java:
```diff
@@ -148,6 +156,13 @@ private void runTest(String inputDates, boolean shouldIncludeModifiedDateInLines
+     * @throws Exception if any occur during testing.
      */
     private void generateReport(String inputDates, boolean shouldIncludeModifiedDateInLines,
                                 boolean shallowCloning, boolean shouldFreshClone,
```
This line and the next line should have the indentation fixed

dcshzj reviewed (changes requested):
Do take note of the additional comments that I pointed out earlier:
1. GitDiff.java at line 28 (violates the additional conventions you proposed)
2. GitLog.java at line 40 and line 57 (violates the additional conventions you proposed)
3. RepoConfiguration.java at line 126 (violates the additional conventions you proposed)
4. CsvParser.java at line 205 (extra newline)
5. CsvParser.java at line 207 (extra space between `return` and `record`)
6. RepoConfigCsvParser.java at line 66 (missing an `@throws InvalidLocationException`)
7. SinceDateArgumentType.java at line 26 (missing an `@throws ArgumentParserException`) [maybe not needed, I'm not sure, I leave this up to you to see if it is necessary]
8. RepoCloner.java at line 118 (violates the additional conventions you proposed)
9. ReportGenerator.java at line 284 (incorrect indentation)

yhtMinceraft1010X commented:
For the proposed convention on no special characters at the start of each line, I realised that it might be difficult to do line-wrapping when a tag happens to be at the very end of a long line. Hence, I'm limiting this new convention to lines in which a complete sentence starts.

dcshzj commented:
@yhtMinceraft1010X Left with this one (updated the line number):
1. ReportGenerator.java at line 312 (incorrect indentation)

dcshzj commented on src/main/java/reposense/parser/GroupConfigCsvParser.java:
```diff
@@ -48,7 +48,7 @@ public GroupConfigCsvParser(Path csvFilePath) throws IOException {
 
     /**
-     * Processes the csv file line by line and adds created {@code Group} into {@code results}.
+     * Processes the csv {@code record} line by line and adds created {@code Group} into {@code results}.
```
Should `Group` be `GroupConfiguration` here?

dcshzj commented on src/main/java/reposense/parser/GroupConfigCsvParser.java:
```diff
@@ -1,6 +1,6 @@
 package reposense.parser;
```
Line 75 of this file, change `Otherwise adds...` to `Otherwise, adds...`.

Also move the `results` variable in line 80 to line 79.

dcshzj commented on src/main/java/reposense/report/RepoCloner.java:
```diff
@@ -269,6 +273,7 @@ private boolean spawnPartialCloneProcess(RepoConfiguration config, boolean shoul
+     * For test environments, cloning is skipped if it has been done before and {@code shouldFreshClone} is false.
      */
     private boolean spawnShallowPartialCloneProcess(RepoConfiguration config, boolean shouldFreshClone) {
         assert(crp == null);
```
The javadoc for `waitForCloneProcess` (in line 313) should include the `config` variable in the javadoc.

dcshzj commented on src/main/java/reposense/report/ReportGenerator.java:
```diff
@@ -266,13 +298,14 @@ private static void setLandingPageTitle(String filePath, String pageTitle) throw
      * successful, and the {@code defaultBranch} of the repo.
      */
     private static CloneJobOutput cloneRepo(RepoConfiguration config, RepoLocation location,
-                                            boolean shouldFreshClone) {
+            boolean shouldFreshClone) {
```
These two lines can be merged into a single line.

dcshzj commented on src/main/java/reposense/report/ReportGenerator.java:
```diff
@@ -285,7 +318,8 @@ private static CloneJobOutput cloneRepo(RepoConfiguration config, RepoLocation l
+     * Uses {@code cloneJobOutput} to find repo location, default branch and whether cloning was successful.
      *
      * @return An {@code AnalyzeJobOutput} object comprising the {@code location} of the repo, whether the cloning was
      * successful, the list of {@code filesGenerated} by the analysis and a list of {@code analysisErrors} encountered.
```
```suggestion
     * successful, the list of {@code generatedFiles} by the analysis and a list of {@code analysisErrors} encountered.
```

dcshzj commented on src/main/java/reposense/report/ReportGenerator.java:
```diff
@@ -348,8 +382,10 @@ private static AnalyzeJobOutput analyzeRepos(String outputPath, List<RepoConfigu
      * @return A list of paths to the JSON report files generated for the repo specified by {@code config}.
+     * @throws NoAuthorsWithCommitsFoundException if there are no authors with commits found for the repo.
      */
     private static List<Path> analyzeRepo(
             RepoConfiguration config, String repoReportDirectory) throws NoAuthorsWithCommitsFoundException {
```
Should move the `config` and `repoReportDirectory` to line 390 and keep only the `throws` portion in line 391.

dcshzj commented on src/main/java/reposense/report/ReportGenerator.java:
```diff
@@ -441,17 +479,17 @@ private static void handleCloningFailed(List<RepoConfiguration> configs, RepoLoc
+     * Adds {@code failedConfigs} that failed cloning/analysis into the list of errors in the summary report along
+     * with an {@code errorMessage} and removes {@code failedConfigs} from the list of {@code configs}.
      */
     private static void handleFailedConfigs(
             List<RepoConfiguration> configs, List<RepoConfiguration> failedConfigs, String errorMessage) {
```
Should move the `configs` and `failedConfigs` variables to line 494, then keep only the `errorMessage` variable in line 495.

dcshzj commented on src/main/java/reposense/report/ReportGenerator.java:
```diff
@@ -482,7 +522,9 @@ private static void handleFailedConfigs(
+     *
      * @return A list of paths to the JSON report files generated for this report.
      */
     private static List<Path> generateIndividualRepoReport(
```
Should move the `repoReportDirectory` variable to line 530 and keep the rest in line 531.

dcshzj commented on src/main/java/reposense/system/LogsManager.java:
```diff
@@ -12,7 +12,7 @@
 import java.util.logging.SimpleFormatter;
```
Should wrap `Console Handler` and `File Handler` in `@code` in the `addConsoleHandler` and `addFileHandler` functions in this file.

dcshzj commented on src/main/java/reposense/util/FileUtil.java:
```diff
@@ -95,8 +95,9 @@ public static void zipFoldersAndFiles(List<Path> pathsToZip,
 
```
Can move `sourcePath` and `outputPath` variables for the `zipFoldersAndFiles` function to line 73, and keep only `fileTypes` in line 74.

dcshzj commented on src/main/java/reposense/util/FileUtil.java:
```diff
@@ -95,8 +95,9 @@ public static void zipFoldersAndFiles(List<Path> pathsToZip,
-     * @return An Optional containing the Path to the JSON file, or an empty Optional
-     *         if there was an error while writing the JSON file.
+     *
+     * @return An Optional containing the Path to the JSON file, or an empty Optional if there was an error
```
`Optional` and `Path` should be wrapped in `{@code}` braces.

dcshzj commented on src/main/java/reposense/util/FileUtil.java:
```diff
@@ -120,8 +121,9 @@ public static void zipFoldersAndFiles(List<Path> pathsToZip,
-     * @return An Optional containing the Path to the ignore revs file, or an empty Optional
-     *         if there was an error while writing the ignore revs file.
+     *
+     * @return An Optional containing the Path to the ignore revs file, or an empty Optional if there was an error
```
`Optional` and `Path` should be wrapped in `{@code}` braces.

dcshzj commented on src/main/java/reposense/util/FileUtil.java:
```diff
@@ -199,7 +204,8 @@ public static void unzip(InputStream is, Path outputPath) throws IOException {
 
     /**
-     * Copies the template files from {@code sourcePath} to the {@code outputPath}.
+     * Copies the template files from {@code is} to the {@code outputPath}.
```
```suggestion
     * Copies the template files from the {@code is} to the {@code outputPath}.
```
Standardize with line 176.

dcshzj commented on src/main/java/reposense/util/TimeUtil.java:
```diff
@@ -65,7 +65,9 @@ public static String getElapsedTimeMessage() {
     /**
-     * Returns a {@code LocalDateTime} that is set to midnight.
+     * Returns a {@code LocalDateTime} that is set to midnight for the given {@code sinceDate}.
+     * If {@code sinceDate} is {@code SinceDateArgumentType.ARBITRARY_FIRST_COMMIT_DATE}, it is simply returned
```
```suggestion
     * If {@code sinceDate} is {@link SinceDateArgumentType#ARBITRARY_FIRST_COMMIT_DATE}, it is simply returned
```

dcshzj commented on src/main/java/reposense/util/TimeUtil.java:
```diff
@@ -137,7 +139,7 @@ public static void verifySinceDateIsValid(LocalDateTime sinceDate, LocalDateTime
     }
```
The `verifySinceDateIsValid` function at line 134 should have the `currentDate` variable in the javadoc.

dcshzj commented on src/main/java/reposense/util/TimeUtil.java:
```diff
@@ -149,8 +151,10 @@ public static String extractDate(String date) {
     /**
-     * Parses the given String as a Date based on the {@code CLI_ARGS_DATE_FORMAT}.
+     * Parses the given {@code date} string as a {@code LocalDateTime} based on the {@code CLI_ARGS_DATE_FORMAT}.
      * Uses {@code ResolverStyle.STRICT} to avoid unexpected dates like 31/02/2020.
```
```suggestion
     * Uses {@link ResolverStyle#STRICT} to avoid unexpected dates like 31/02/2020.
```

dcshzj commented on src/systemtest/java/reposense/ConfigSystemTest.java:
```diff
@@ -148,10 +156,17 @@ private void runTest(String inputDates, boolean shouldIncludeModifiedDateInLines
+     * @param shallowCloning Boolean for whether to perform shallow cloning.
+     * @param shouldFreshClone Boolean for whether to clone repo again if it has been cloned before.
+     * @param findPreviousAuthors Boolean for whether to find and blame previous authors for ignored commits.
+     * @throws Exception if any occur during testing.
```
```suggestion
     * @throws Exception if any errors occur during testing.
```

dcshzj commented on src/systemtest/java/reposense/ConfigSystemTest.java:
```diff
@@ -148,10 +156,17 @@ private void runTest(String inputDates, boolean shouldIncludeModifiedDateInLines
     private void generateReport(String inputDates, boolean shouldIncludeModifiedDateInLines,
-                                boolean shallowCloning, boolean shouldFreshClone,
-                                boolean findPreviousAuthors) throws Exception {
+            boolean shallowCloning, boolean shouldFreshClone,
+            boolean findPreviousAuthors) throws Exception {
```
These two lines can be merged.

dcshzj commented on src/test/java/reposense/model/RepoLocationTest.java:
```diff
@@ -47,7 +47,8 @@ public void repoLocationParser_parseEmptyString_success() throws Exception {
+     * Compares the information of {@code rawLocation} parsed by the RepoLocation model with {@code expectedRepoName}
+     * and {@code expectedOrganization}.
      */
     public void assertValidLocation(String rawLocation, String expectedRepoName,
             String expectedOrganization) throws Exception {
```
Should move the `expectedOrganization` variable to line 52 and keep the `throws Exception` in line 53.

dcshzj commented on src/test/java/reposense/util/TestUtil.java:
```diff
@@ -223,7 +225,7 @@ public static boolean isWindows() {
     }
```
Line 196 in this file, should change `{@code set}` to `{@code Set}`.

dcshzj reviewed (changes requested):
Might need to change some instances of `@code` to `@link`.

yhtMinceraft1010X commented:
Re-requesting review from @reposense/active-developer since a lot of changes have been made, especially with a few merged PRs

chan-j-d commented:
Thanks for doing this up. Was wondering this https://github.com/reposense/RepoSense/pull/1688#issuecomment-1049877487 in consideration, is there a consensus to using `@link` when mentioning imported classes? This seems like quite a significant change for code maintenance and I'm not too sure if the benefits are significant since I can already hover over the variable itself now:
![image](https://user-images.githubusercontent.com/65345505/155856306-4dedad38-5b12-467e-a65f-00d1ed4614f4.png)

chan-j-d commented on src/main/java/reposense/commits/CommitResultAggregator.java:
```diff
@@ -55,7 +56,9 @@ public static CommitContributionSummary aggregateCommitResults(
 
     /**
-     * Calculates the contribution variance of all authors.
+     * Calculates the contribution variance of all authors across contributions made between {@code startDate}
```
I think the first sentence can retain the original `Calculates the contribution variance of all authors within a date range.` All the remaining details can follow in additional sentences.

chan-j-d commented on src/main/java/reposense/commits/CommitResultAggregator.java:
```diff
@@ -104,6 +112,14 @@ private static float getContributionVariance(List<AuthorDailyContribution> contr
     }
 
+    /**
+     * Creates a {@link Map} of each author in {@code authorSet} to their respective {@code commitResults}
```
I think the first sentence here can be `Returns a mapping of each {@link author} to their commit contributions.` The rest of it seems to go quite into detail on how they are represented which can be placed in the second sentence and onwards.

chan-j-d commented on src/main/java/reposense/commits/CommitResultAggregator.java:
```diff
@@ -148,8 +167,9 @@ private static LocalDateTime getStartOfDate(LocalDateTime current, ZoneId zoneId
-     * the {@code ARBITRARY_FIRST_COMMIT_DATE} converted to the timezone given by {@code zoneId}.
+     *
+     * @return First commit date if there is at least one {@link CommitResult}. Otherwise, return
+     * the {@code ARBITRARY_FIRST_COMMIT_DATE_UTC} converted to the timezone given by {@code zoneId}.
```
should be `{@link Class#ARBITRARY_FIRST_COMMIT_DATE_UTC}`?

yhtMinceraft1010X replied:
This shows a non-working red link. I also realised that ARBITRARY_FIRST_COMMIT_DATE_UTC is a private constant, so it may not be worth mentioning directly.

chan-j-d commented on src/main/java/reposense/git/GitRevList.java:
```diff
@@ -19,8 +19,9 @@
     /**
-     * Returns the latest commit hash before {@code date}, with {@code ZoneId} taken into account.
-     * Returns an empty {@code String} if {@code date} is null, or there is no such commit.
+     * Returns the latest commit hash at the branch given by {@code branchName} before {@code date}, with
```
```suggestion
     * Returns the latest commit hash at {@code branchName} before {@code date}. ...
```

chan-j-d commented on src/main/java/reposense/git/GitRevList.java:
```diff
@@ -52,8 +54,10 @@ public static String getCommitHashUntilDate(String root, String branchName, Loca
     /**
-     * Returns a list of commit hashes separated by newline that are within the range of {@code startHash} and
-     * {@code endHash}. Both the {@code startHash} and {@code endHash} are guaranteed to be in the list.
+     * Returns a list of commit hashes at the branch given by {@code branchName}, separated by newlines,
```
```suggestion
     * Returns a list of commit hashes at {@code branchName} between {@code startHash} and {@code endHash}. ...
```

chan-j-d reviewed (changes requested):
I only managed to review about half of the line changes as the PR is pretty large. I think it might have been ideal to split one package per PR. 

My main concern is that quite a few of the Javadocs have extremely long first sentences. This does not follow the standard described in SE-EDU since it is used as a summary for the method when converted into Javadocs.

chan-j-d commented on src/main/java/reposense/model/RepoConfiguration.java:
```diff
@@ -229,22 +230,26 @@ public static void setFormatsToRepoConfigs(List<RepoConfiguration> configs, List
         }
     }
 
+    /**
+     * Checks if there are any repos in {@code configs} for which RepoSense is trying to blame line changes
+     * caused by commits in the ignore commit list to any available previous authors who changed those lines.
+     */
```
I personally don't think this method requires a Javadocs. I think it falls under the self-explanatory category where we're checking if a specific boolean flag is ticked. 

If necessary, I think something like
```suggestion
    /**
     * Checks if any of the {@code configs} is finding previous authors for commit analysis.
     */
```

chan-j-d commented on src/main/java/reposense/model/RepoConfiguration.java:
```diff
@@ -268,8 +273,8 @@ public void update(StandaloneConfig standaloneConfig) {
     /**
-     * Attempts to find matching {@code Author} given a name and an email.
-     * If no matching {@code Author} is found, {@code Author#UNKNOWN_AUTHOR} is returned.
+     * Attempts to find matching {@link Author} given a {@code name} and an {@code email}.
```
```suggestion
     * Returns the matching {@link Author} given a {@code name} and an {@code email}.
```

chan-j-d commented on src/main/java/reposense/parser/CsvParser.java:
```diff
@@ -171,9 +176,9 @@ protected String getOrDefault(final CSVRecord record, String header, String defa
 
     /**
-     * Returns the value of {@code record} at the column with the header {@code header} as a {@code List},
+     * Returns the value of {@code record} at the column with the header {@code header} as a {@link List},
```
I think this line can end with a full stop. The delimiter and empty list behaviour can be in follow-up sentences.

chan-j-d reviewed (changes requested):
I've looked through the remainder of the methods in the PR with this. Most of it looks great. 

My only concern is with the standardization of linking standard Java classes. I think something like `LocalDateTime` is appropriate to link but classes like `String` or `Integer` seem somewhat redundant to link. I would like to know your thoughts on this and if it should be standardized.

yhtMinceraft1010X commented on src/main/java/reposense/git/GitUtil.java:
```diff
@@ -57,7 +57,7 @@ static String convertToGitDateRangeArgs(LocalDateTime sinceDate, LocalDateTime u
 
     /**
-     * Returns the {@code String} command to specify the authors to analyze for `git log` command.
+     * Returns the {@link String} command to specify the {@code author} to analyze for `git log` command.
```
I'm currently splitting the PR into packages, so I'll incorporate any new changes in the PR for the `git` package.

yhtMinceraft1010X replied:
Sounds like a good point though. String is quite commonly used so there's not really a need to link it.

yhtMinceraft1010X commented:
Pull request is now split into different pull requests for each package. This current pull request will be kept open to incorporate checkstyle and documentation changes but all the split pull requests must be merged first.

dcshzj commented:
> Pull request is now split into different pull requests for each package.

I personally do not agree with doing this, especially since quite a number of reviews has been done on this PR. We have got a number of huge PRs before, and while it is discouraged, I preferred to have kept everything in this PR because we are also looking to standardize the conventions across the entire codebase.

gerhean commented:
The reason for splitting up large PR is to ensure that changes to logic is more finely tested and hence easier to catch and rollback. However since this is just a documentation change, there should be no change in code behavior, hence no need to do partial rollbacks and hence not much need to split up this PR.

yhtMinceraft1010X commented:
One of the concerns mentioned during the CS3281 meeting earlier today is that there is too much code to review, which could slow down the PR's progress. Hence my decision to split the PR into packages (+ do a separate housekeeping PR for fixes to the code style in the actual code). 

The link to the proposed Javadoc style guide will be provided in the descriptions for the new PRs.

damithc commented:
@yhtMinceraft1010X Sorry if my comment earlier in the meeting caused you more work. The important thing is the cognitive load imposed on the reviewer rather than the number of files/lines changed. Note,
* Having some reviews already in the PR is a disincentive to split up. 
* If splitting up a future PR of similar nature, the split can also be based on the change rather than file/package e.g., `add missing blank line in header comment`. That way, reviewer has to check for only one thing and s/he can also ensure the consistency across the code base.

dcshzj commented:
Sorry @yhtMinceraft1010X are you able to close the PRs that you created? We can keep #1692 since there are already reviews there, and continue the rest of the reviews here

yhtMinceraft1010X commented:
I've re-consolidated all the PRs except for #1692, which I feel is best kept separate.
> Also, try to avoid merging updates to the `master` branch unless there are conflicts to be resolved. It causes a lot of CI jobs to be triggered and we can potentially be flagged by GitHub for spam.
Will do.

chan-j-d commented on src/main/java/reposense/git/GitRevList.java:
```diff
@@ -19,9 +19,13 @@
-     * with {@code ZoneId} taken into account.
-     * Returns an empty {@code String} if {@code date} is null, or there is no such commit.
+     * Returns the latest commit hash at {@code branchName} before {@code date}.
+     * Returns an empty {@link String} if {@code date} is null, or there is no such commit.
```
If we're standardizing to `{@code String}` then there are couple of instances of leftover `{@link String}`

chan-j-d commented on src/main/java/reposense/parser/AlphanumericArgumentType.java:
```diff
@@ -8,7 +8,7 @@
 
 /**
- * Represents an alphanumeric type String argument.
+ * Represents an alphanumeric type {@link String} argument.
```
change to `{@code String}`?

chan-j-d commented on src/main/java/reposense/parser/CsvParser.java:
```diff
@@ -112,11 +115,13 @@ public CsvParser(Path csvFilePath) throws IOException {
-     * The line is split into an array of Strings, using the comma symbol as delimiter.
+     * Returns the header of a CSV file, which is assumed to be the first non-empty / non-whitespace line in the file
+     * read by {@code reader}.
+     * The line is split into an array of {@link String}s, using the comma symbol as delimiter.
```
change to `{@code String}`?

chan-j-d reviewed (changes requested):
@yhtMinceraft1010X Most of the changes that I have requested have been addressed. Thanks for the hard work. Besides the remaining `{@link String}` mentions, there are also two comments on the `styleGuides.md` from a previous review which I would like to know your thoughts on.

chan-j-d reviewed (approved):
LGTM!

yhtMinceraft1010X commented:
Should we close #1202 after merging this?

dcshzj commented on src/main/java/reposense/report/ReportGenerator.java:
```diff
@@ -98,7 +98,20 @@
+     * @param isSinceDateProvided The boolean variable for whether client provided a sinceDate.
+     * @param isUntilDateProvided The boolean variable for whether client provided an untilDate.
+     * @param numCloningThreads The number of threads to use to clone the repos.
+     * @param numAnalysisThreads The number of threads to use to analyse the repos.
```
```suggestion
     * @param numAnalysisThreads The number of threads to use to analyze the repos.
```

dcshzj commented on src/main/java/reposense/report/ReportGenerator.java:
```diff
@@ -114,6 +127,20 @@
+     * @param isSinceDateProvided The boolean variable for whether client provided a sinceDate.
+     * @param isUntilDateProvided The boolean variable for whether client provided an untilDate.
+     * @param numCloningThreads The number of threads to use to clone the repos.
+     * @param numAnalysisThreads The number of threads to use to analyse the repos.
```
```suggestion
     * @param numAnalysisThreads The number of threads to use to analyze the repos.
```

dcshzj commented on src/main/java/reposense/model/CommitHash.java:
```diff
@@ -79,14 +80,15 @@ public int hashCode() {
     }
```
Still got this comment, need to include `root` and `branchName` in the Javadoc for the getHashes function.

dcshzj commented on src/main/java/reposense/git/GitShortlog.java:
```diff
@@ -39,8 +39,12 @@
-    private static String getShortlogSummary(String root, LocalDateTime sinceDate,
-            LocalDateTime untilDate, ZoneId zoneId) {
+    /**
+     * Obtains summarised version of git log from the repository at {@code root} for the date range
```
```suggestion
     * Obtains summarized version of git log from the repository at {@code root} for the date range
```

dcshzj reviewed (changes requested):
Just left 4 changes.

dcshzj reviewed (approved):
LGTM! Great work on this PR!

yhtMinceraft1010X commented:
I would like to request for #1652 to be merged before this one since there are some new Javadoc checks there and I'm not sure if there are any Javadoc that will fail those new checks.

dcshzj commented:
> I would like to request for #1652 to be merged before this one since there are some new Javadoc checks there and I'm not sure if there are any Javadoc that will fail those new checks.

@yhtMinceraft1010X I have merged #1652.

yhtMinceraft1010X commented:
I have resolved the merge conflicts and also added a new check in checkstyle for validating throws tags. Since the new check resulted in some Javadoc failing it, I have also added the `@throws` tags for the failing checks.

dcshzj commented on config/checkstyle/suppressions.xml:
```diff
@@ -6,4 +6,6 @@
 
 <suppressions>
   <suppress checks="JavadocType" files=".*Test\.java"/>
+  <suppress checks="JavadocMethod" files=".*[\\/]src[\\/]test[\\/]"/>
+  <suppress checks="JavadocMethod" files=".*[\\/]src[\\/]systemtest[\\/]"/>
```
Is it possible to suppress the JavadocMethod check for functions with the `@Test` annotation present? The `@Before`, `@After` and other test-related annotations can also be suppressed.

dcshzj replied:
@yhtMinceraft1010X

yhtMinceraft1010X replied:
The check is already suppressed for the test-related annotations under `MissingJavadocMethod` in the `checkstyle.xml` file. The reason why I put these suppressions is to allow for missing `@throws` tags in the non-annotated functions for test code since my proposed style guide only calls for `@throws` tags in main code.

dcshzj replied:
I see, okay sure. I think it should be sufficient for now, although I think in the future we can try to generalize the style guide to cover all aspects of the code base and to maintain consistency.
